examples/handlers/lambda.links

```diff
@@ -60,7 +60,7 @@ fun lit(x)() { x }
 
 # Due to the (current) lack of multi-handlers in Links, we are going
 # simulate them by reifying operations.
-typename Expr(a) = Comp(a, {Var:(Var) -> Expr(a), Lam:(Var) -> ()});
+typename Expr(a) = Comp(a, {Var:(Var) {}-> Expr(a), Lam:(Var) {}-> ()});
```

Comment:
Why are the operation arrows explicitly closed? The existing sugar should rewrite `(Var) -> Expr(a)` as `(Var) {}-> Expr(a)`. 

Ideally, one should not be able to write `{}->` at all, since the arrow in question is not a function arrow, but that is an entirely different concern.

Reply:
The trouble is that our check for "does this typename require an implicit effect variable" occurs before the desugaring has run, and so it thinks this is a normal arrow.

It's a bug (hence me adding the test case), I just wanted to get something working - it's on my list to fix tomorrow, I just need to think about how best to do it.